application/controllers/Vendas.php

```diff
@@ -295,6 +301,14 @@ public function adicionarProduto()
         $this->form_validation->set_rules('idProduto', 'Produto', 'trim|required');
         $this->form_validation->set_rules('idVendasProduto', 'Vendas', 'trim|required');
 
+        $editavel = $this->vendas_model->isEditable($this->input->post('idVendasProduto'));
+        if (!$editavel) {
+            return $this->output
+                ->set_content_type('application/json')
+                ->set_status_header(200)
+                ->set_output(json_encode(['result' => false, 'messages'=> '<br /><br /> <strong>Motivo:</strong> Venda já faturada']));
+        }
```

Comment:
Não retornar status code 200 quando for erro. Recomendo usar 400 ou 422.

Reply:
corrigido